### PR TITLE
[risk=no][RW-6298] Address some tech debt in the account survey

### PR DESCRIPTION
**Pull request by petesantos**
Description:

Factored the error modal out of the survey, utilizing some existing error handling/modal code.
The advantages of this are:
1. Removes the need for bespoke error modals for the account and update
2. Removes state from the account and profile components
3. This will help remove test code, allowing for better test consolidation 
4. Paves the way for further simplification of some of the survey related code

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/components/text-modal.tsx [hunk 1]
```diff
@@ -20,6 +20,7 @@ export interface TextModalProps {
   body: string;
   buttonText: string;
   closeFunction: Function;
+  role?: 'dialog' | 'alertdialog';
```
Comment (jmthibault79) [inline ui/src/app/components/text-modal.tsx:23]: Is this an ADT?  very cool.  How does this compare to using an Enum?

File: ui/src/app/components/with-error-modal.tsx [hunk 2]
```diff
@@ -35,14 +36,44 @@ export const withErrorModal = () => {
...
       render() {
         return <React.Fragment>
-          {this.state.show && <TextModal title={this.state.title}
+          {this.state.show && <TextModal role='alertdialog'
+                                         title={this.state.title}
                                          body={this.state.body}
                                          closeFunction={() => this.setState({show: false})}/>
           }
-          <WrappedComponent showErrorModal={(tile, body) => this.show(tile, body)}
+          <WrappedComponent showErrorModal={(title, body) => this.show(title, body)}
```
Comment (jmthibault79) [inline ui/src/app/components/with-error-modal.tsx:44]: ha

File: ui/src/app/components/with-error-modal.tsx [hunk 3]
```diff
@@ -35,14 +36,44 @@ export const withErrorModal = () => {
...
   };
 };
+
+export interface WithProfileErrorModalProps {
+  showProfileErrorModal?: (message: string) => void;
+}
+
+export const withProfileErrorWrapper = ({title = ''}) => {
+  const body = ({message}) => (<React.Fragment>
+    <div>An error occurred while saving profile. The following message was
```
Comment (jmthibault79) [inline ui/src/app/components/with-error-modal.tsx:58]: how about "your profile"

File: ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import {mount, ReactWrapper} from 'enzyme';
 import * as React from "react";
 
 import {serverConfigStore} from 'app/utils/navigation';
+// import {WithProfileErrorModalProps} from 'app/components/with-error-modal';
```
Comment (jmthibault79) [inline ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx:5]: rm

**Review by jmthibault79 — APPROVED** [review]
This is really nice, thanks.  My only concern is the specifics of the error text - maybe run it by Shimon?

File: ui/src/app/components/with-error-modal.tsx (hunk 2 above)
Comment (dmohs) [reply ui/src/app/components/with-error-modal.tsx:44]: Could this be,
```jsx
<WrappedComponent showErrorModal={this.show.bind(this)} ...>
```
to eliminate the potential for a spelling error in the first place?

Comment (dmohs) [reply ui/src/app/components/with-error-modal.tsx]: Indeed, this seems to work.

File: ui/src/app/components/with-error-modal.tsx (hunk 3 above)
Comment (dmohs) [reply ui/src/app/components/with-error-modal.tsx]: 👍

File: ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx (hunk 4 above)
Comment (dmohs) [reply ui/src/app/pages/login/account-creation/account-creation-survey.spec.tsx]: 👍

File: ui/src/app/components/text-modal.tsx (hunk 1 above)
Comment (dmohs) [reply ui/src/app/components/text-modal.tsx:23]: Not sure, please ask offline so we can discuss.

**Review by dmohs — APPROVED** [review]

**Pull request merged**